libsplinter/src/runtime/service/timer/thread.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you mcay not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The internal thread used by the Timer
+
+use std::sync::mpsc::Receiver;
+use std::thread;
+
+use crate::error::InternalError;
+use crate::service::{FullyQualifiedServiceId, MessageSenderFactory};
+use crate::threading::pool::{JobExecutor, ThreadPoolBuilder};
+
+use super::message::TimerMessage;
+use super::FilterCollection;
+
+const DEFAULT_POOL_SIZE: usize = 8;
+
+pub struct TimerThread {
+    join_handle: thread::JoinHandle<()>,
+}
+
+impl TimerThread {
+    pub fn start(
```

Comment:
The inner thread doesn't follow the two-phase threading::ShutdownHandle approach when handling the thread_pool.

Reply:
Updated